stetho/src/main/java/com/facebook/stetho/inspector/helper/ChromePeerManager.java

```diff
@@ -94,9 +102,12 @@ public synchronized boolean hasRegisteredPeers() {
     return !mReceivingPeers.isEmpty();
   }
 
-  private synchronized ArrayList<JsonRpcPeer> getReceivingPeersCopy() {
-    // This performance should be OK in practice because we rarely have more than 1 peer connected.
-    return new ArrayList<JsonRpcPeer>(mReceivingPeers.keySet());
+  private synchronized JsonRpcPeer[] getReceivingPeersSnapshot() {
+    if (mReceivingPeersSnapshot == null) {
```

Comment:
Oops, accidentally commented on your commit before.  Moving here: prefer this in a separate commit (can be the same PR tho).  It seems unrelated generally.